plugins/kubernetes/app/models/kubernetes/deploy_yaml.rb

```diff
@@ -46,6 +47,46 @@ def template
       end
     end
 
+    # Sets up the secret_sidecar and the various mounts that are required
+    # if the sidecar service is enabled
+    # /vaultauth is a secrets volume in the cluster
+    # /secretkeys are where the annotations from the config are mounted
+    def set_secret_sidecar
+      pod_volumes =
+        [
+          {name: "secrets-volume", emptyDir: {}},
+          {name: "vaultauth", secret: {secretName: "vaultauth"}},
+          {name: "secretkeys", downwardAPI: {items: [{path: "annotations", fieldRef:{fieldPath:"metadata.annotations"}}]}}
+        ]
+      secret_vol = { mountPath: "/secrets", name: "secrets-volume" }
+      secret_sidecar = {
+        image: ENV.fetch("SECRET_SIDECAR_IMAGE").to_s,
+        name: "secret-sidecar",
+        volumeMounts: [
+          secret_vol,
+          { mountPath: "/vault-auth", name: "vaultauth" },
+          { mountPath: "/secretkeys", name: "secretkeys" }
+        ]
+      }
+      #TODO, do we want resource limits on this, or just give it what it needs?
```

Comment:
should need next to nothing ? ... maybe deploy on it's own and see what the minimum is ... 100m seems like too much for a simple script ... 
ideally we'd boot it and then shut it down since it will not be needed again ... maybe even better if we can make it a daemonset so we don't need 1 per pod ... but having something that actually works is a good first step :)